packages/swatch/src/SwatchGroup.ts

```diff
@@ -289,19 +308,34 @@ export class SwatchGroup extends SizedMixin(SpectrumElement, {
             });
         }
 
-        // Do Swatch actions to each Swach in the collection.
-        this.rovingTabindexController.elements.forEach((swatch) => {
-            swatchActions.forEach((action) => {
-                action(swatch);
+        const doActions = (): void => {
+            nextSelected = new Set(this.selected);
+
+            // Do Swatch actions to each Swach in the collection.
+            this.swatches.forEach((swatch) => {
+                swatchActions.forEach((action) => {
+                    action(swatch);
+                });
             });
-        });
 
-        // Apply state built in actions back to the Swatch Group
-        if (changes.has('selected')) {
-            this.selected = [...nextSelected].filter((selectedValue) =>
-                currentValues.has(selectedValue)
+            // Apply state built in actions back to the Swatch Group
+            if (changes.has('selected')) {
+                this._selected = [...nextSelected.values()].filter(
+                    (selectedValue) => currentValues.has(selectedValue)
+                );
+            }
+        };
+
+        if (this.hasUpdated) {
+            doActions();
+        } else {
+            this.shadowRoot.addEventListener(
+                'slotchange',
+                () => {
+                    requestAnimationFrame(doActions);
+                },
+                { once: true }
             );
```

Comment:
The branch here is on whether or not the component has updated at least once, but if not we listen for `slotchange` instead of update. It makes me wonder if this would be equivalent (if not, a comment might be helpful down the line).

```ts
if (!this.hasUpdated) {
    await this.updateComplete;
}
doActions();
```